hcloud/image.go

```diff
@@ -91,22 +91,12 @@ func (c *ImageClient) GetByID(ctx context.Context, id int) (*Image, *Response, e
 
 // GetByName retrieves an image by its name. If the image does not exist, nil is returned.
 func (c *ImageClient) GetByName(ctx context.Context, name string) (*Image, *Response, error) {
-	path := "/images?name=" + url.QueryEscape(name)
-	req, err := c.client.NewRequest(ctx, "GET", path, nil)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	var body schema.ImageListResponse
-	resp, err := c.client.Do(req, &body)
-	if err != nil {
-		return nil, nil, err
+	images, response, err := c.List(ctx, ImageListOpts{Name: name})
+	var image *Image
+	if len(images) > 0 {
+		image = images[0]
```

Comment:
If you’d check for `len(images) == 0` and return then immediately, you could skip the temporary variable and make the code easier to read (at least to me).